lib/models/application.coffee

```diff
@@ -341,13 +341,16 @@ getApplicationModel = (deps, opts) ->
 		else
 			Promise.resolve()
 
-		deviceSlugPromise = deviceModel().getDeviceSlug(deviceType)
+		deviceManifestPromise = deviceModel().getManifestBySlug(deviceType)
 		.tap (deviceSlug) ->
 			if not deviceSlug?
 				throw new errors.ResinInvalidDeviceType(deviceType)
 
-		return Promise.all([ deviceSlugPromise, parentAppPromise ])
-		.then ([ deviceSlug, parentApplication ]) ->
+		return Promise.all([ deviceManifestPromise, parentAppPromise ])
+		.then ([ deviceManifest, parentApplication ]) ->
+			if deviceManifest.state == 'DISCONTINUED'
```

Comment:
How about adding
```coffee
throw new errors.ResinInvalidDeviceType(deviceType) if !deviceManifest
```
?

Reply:
I think we discussed this at summit, and it is already covered by the `.tap` check above.

Reply:
Opps, forgot to approve right after!